File: CS50/Week1/credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
        long n = get_long(" Number:");

        long m = n;

        int i = 0;

        int mult = 0;

        int sum = 0;

        for (i = 0; n > 0; i++)
        {

                int digit = n % 10;

                // somar pares

                if (i % 2 == 0)
                {
                        sum = sum + digit;
                }

                else
                {
                        // Multiplicar impares
                        if (digit * 2 >= 10)
                        {

                                int aux = (digit * 2) % 10;

                                int aux2 = (digit * 2) / 10;

                                mult = mult + aux + aux2;
                        }

                        else
                        {

                                mult = mult + digit * 2;
                        }
                }
                n = n / 10;
        }

        if ((sum + mult) % 10 == 0)
        {

                do
                {
                        m = m / 10;
                }

                while (m > 100);
                // i identifica os digitos do numero
                if ((m == 34 || m == 37) && i == 15)
                {

                        printf("AMEX\n");
                }

                else if ((m == 51 || m == 52 || m == 53 || m == 54 || m == 55) && i == 16)
                {

                        printf("MASTERCARD\n");
                }

                else if (m / 10 == 4 && (i == 13 || i == 16))

                {

                        printf("VISA\n");
                }
                else
                {

                        printf("INVALID\n");
                }
        }

        else
        {

                printf("INVALID\n");
        }
}

File: CS50/Week6/sentimental-credit/credit.py
from cs50 import get_int

n = get_int("Numero do cartao: ")
m = n
b = m
mult = 0
soma = 0

i = 1

while n > 0:

    digit = n % 10

    if i % 2 == 0:

        aux1 = digit * 2

        if aux1 > 9:

            aux2 = (aux1 % 10) + (aux1 // 10)

        else:

            aux2 = aux1

        mult = mult + aux2

    elif i % 2 != 0:

        soma = soma + digit

    i += 1

    n = int(n) // 10

if (mult + soma) % 10 != 0:

    print("INVALID")

else:

    while m > 100:

        m = m // 10

    if m == 34 or m == 37 and len(str(b)) == 15:

        print("AMEX")

    elif m == 51 or m == 52 or m == 53 or m == 54 or m == 55 and len(str(b)) == 16:

        print("MASTERCARD")

    elif m // 10 == 4 and len(str(b)) == 13 or len(str(b)) == 16:

        print("VISA")

    else:

        print("INVALID")